Feature/pages transitions

AlexandreXYZ opened the pull request:
Add pages transition for React-routes

AlexandreXYZ committed:
test(client): add "history" from URL modify

History is a library from "ReactTraining" repository that allows us modify the website URL

BREAKING CHANGE: error in app.tsx because of a useEffect (comment to resolve)

AlexandreXYZ committed:
test(client): history does not stack URL and does not overlap the previous

AlexandreXYZ committed:
style(client): rename function from components "addQuest" and "addResponse"

Two functions that had the same names 'AddCard' have been renamed

AlexandreXYZ committed:
refactor(client): tests with the 'history' library finished

The 'history' library has been tested and concluded that it will not be used in the project, so it
is being removed and React-Router-Dom has been added to replace it.

AlexandreXYZ committed:
build(client): added and applied the 'react-router-dom' library

This library provided a simple and practical URL screen transition for components and their
children.

ViniciusGregorine commented on client/src/App.tsx:
```diff
@@ -2,11 +2,35 @@ import './css/App.css'
+// let a = new history('test');
+
 export default function App() {
+	const [getDashbortType, setDashboardType] = useState('Dashboard')
```
**possibilidade** da retirada disso

AlexandreXYZ replied:
Resolvido no commit 49e7046

ViniciusGregorine reviewed (commented):
"ta no caminho"

AlexandreXYZ committed:
refactor(client): some components have been merged and useless conditions have been deleted

Some imports and codes that were not being used were removed and the revisionquest component was
merged with revisionresponse and the addquest with addresponse generating the Revision and add
files.

AlexandreXYZ committed:
style(client): transform ADD function in arrow function

AlexandreXYZ committed:
style(client): add tag Router in Index.tsx

AlexandreXYZ committed:
feat(client): page changes by buttons enabled

AlexandreXYZ committed:
style(client): modified small code details

AlexandreXYZ committed:
fix(client): rename Class and IDs

AlexandreXYZ committed:
style(client): add extend in .AddCard

AlexandreXYZ committed:
style(client): move some ids and classes from app.sass

Moved hit, miss and MissAndHit classes and ids from app.scss to Revision.scss

k1vz commented on client/src/components/Add.tsx:
```diff
@@ -0,0 +1,33 @@
+    return(
+        <>
+        <div className="AddCard">
+            { side === 'front' && <div id="AddContainer_title">
```
dá uma revisada nessa identação das chaves aq, tipo coloca igual as chaves de 'if' tlg

AlexandreXYZ replied:
Você diz tipo:
```
side === 'front' && 
<div>...</div>
```
?

k1vz replied:
tipo
```
{
side === 'front' && <div id="AddContainer_title">
```

k1vz commented on client/src/components/Revision.tsx:
```diff
@@ -1,17 +1,19 @@
 }
 
-export default function Revision({dataCards, buttonIndex}: Props) {
+const Revision = ({dataCards, buttonIndex,}: Props) => {
```
tem uma vírgula a mais ali

AlexandreXYZ replied:
kkkkkkkkkk vlw

k1vz commented on client/src/css/AddCard.css:
```diff
@@ -9,42 +9,98 @@
   background-color: #ebf5ff; }
```
fecha chaves precisa de atenção dnv

AlexandreXYZ replied:
Os arquivos css são pré processados pelo SASS então isso não fui eu que fiz, é um resuldado dos meus codigos SASS.

k1vz commented on client/src/css/Revision.css:
```diff
@@ -31,8 +31,10 @@
   height: 3rem;
   position: relative; }
-  .card_footer > div {
+  .card_footer > a {
```
'a'?

AlexandreXYZ replied:
kkkkkkkkk é a tag html Link (um component do react-router-dom) que quando é transfomado em HTML vira uma tag "a" de ancora.

k1vz commented on client/src/index.tsx:
```diff
@@ -3,10 +3,13 @@ import ReactDOM from 'react-dom'
 import './css/index.css'
 import App from './App'
 import reportWebVitals from './reportWebVitals'
+import { BrowserRouter as Router} from "react-router-dom";
```
faltou um espaço dps do Router

k1vz commented on client/src/index.tsx:
```diff
@@ -3,10 +3,13 @@ import ReactDOM from 'react-dom'
+    <Router>
+      <App/>
+    </Router>
   </React.StrictMode>,
```
Tem uma vírgula ali no fim

AlexandreXYZ replied:
Se tu perceber não foi uma linha editada ksksksksks mas é pq ele não é um component como a gente usa tipo:
```
Return(
<div>...</div>
)
```
Mas é p ReactDOM.render, então ele tem algumas tag de config tipo o StrictMode, Router... mas se ver tem JS tambem que é oq vem depois da virgura:
```
ReactDOM.render(
  <React.StrictMode>
    <Router>
      <App/>
    </Router>
  </React.StrictMode>,
  document.getElementById('root') 
)
```
Resumindo: se tirar dá merda :)

ViniciusGregorine reviewed (approved):
all good

k1vz commented on client/src/pages/Dashboard.tsx:
```diff
@@ -23,19 +21,18 @@ const Dashboard = () => {
-
+                <Switch>
+                    <Route exact path={`/dashboard/front`}>
+                        <RevisionQuest dataCards={orderCards} buttonIndex={index}/>
```
aq ainda ta usando como 'quest' e 'response', n seria interessante mudar pra 'front' e 'back' tb?

AlexandreXYZ replied:
Isso é só mesmo algo de interpretação no HTML, pra ti saber qual tag é qual, não faz diferença.

k1vz reviewed (changes requested):
Tá ótimo, só dá uma olhada nos comentários antes de dar merge

AlexandreXYZ committed:
style(client): resolved changes requested by Vinicius and Pedro
